control-plane-operator/hostedclusterconfigoperator/controllers/resources/resources.go

```diff
@@ -1389,6 +1416,65 @@ func (r *reconciler) reconcileClusterVersion(ctx context.Context, hcp *hyperv1.H
 	return nil
 }
 
+func (r *reconciler) reconcileControlPlaneDataPlaneConnectivityConditions(ctx context.Context, hcp *hyperv1.HostedControlPlane, log logr.Logger) error {
+	condition := &metav1.Condition{
+		Type:    string(hyperv1.ControlPlaneToDataPlaneConnectivityHealthy),
+		Status:  metav1.ConditionUnknown,
+		Reason:  hyperv1.ControlPlaneToDataPlaneReasonNoWorkerNodesAvailable,
+		Message: "No worker nodes available",
+	}
+	totalNodes, err := util.CountAvailableNodes(ctx, r.client)
+	if err != nil {
+		condition.Message = "Unable to count worker nodes: " + err.Error()
+		log.Error(err, "Unable to count worker nodes")
+	}
+	if totalNodes > 0 {
+		condition.Status = metav1.ConditionFalse
+		condition.Reason = hyperv1.ControlPlaneToDataPlaneReasonNoKonnectivityAgentPodsFound
+		condition.Message = "Couldn't find any konnectivity-agent running in data plane"
+		var podList corev1.PodList
+		if err := r.uncachedClient.List(ctx, &podList,
+			client.MatchingLabels{"app": "konnectivity-agent"}, client.InNamespace("kube-system")); err != nil {
+			log.Error(err, "unable to list konnectivity-agent PODs in kube-system namespace")
+		}
+		for _, pod := range podList.Items {
+			if pod.Status.Phase != corev1.PodRunning {
+				continue
+			}
+			condition.Status = metav1.ConditionFalse
+			condition.Reason = hyperv1.ControlPlaneToDataPlaneReasonLogAccessFailed
+			data, err := r.GetPodLogs(ctx, r.clientSet, pod.Namespace, pod.Name, "konnectivity-agent")
+			if err == nil && len(data) > 0 { // found something
+				condition.Status = metav1.ConditionTrue
+				condition.Reason = hyperv1.ControlPlaneToDataPlaneReasonConnectivityOK
```

Comment:
please use `hyperv1.AsExpectedReason` when the condition is true

Reply:
OK